src/main/java/com/google/devtools/build/lib/rules/cpp/CcLinkingHelper.java

```diff
@@ -857,6 +871,9 @@ private CppLinkActionBuilder newLinkActionBuilder(
    */
   private Artifact getLinkedArtifact(LinkTargetType linkTargetType) throws RuleErrorException {
       String maybePicName = label.getName() + linkedArtifactNameSuffix;
+      if(linkTargetType == LinkTargetType.NODEPS_DYNAMIC_LIBRARY) {
+        maybePicName += linkedDLLNameSuffix;
+      }
```

Comment:
Can we move this to line 882? Like
```
String linkedName = maybePicName + (linkTargetType == LinkTargetType.NODEPS_DYNAMIC_LIBRARY ? linkedDLLNameSuffix : "");
linkedName =
    CppHelper.getArtifactNameForCategory(
              ruleErrorConsumer, ccToolchain, linkTargetType.getLinkerOutput(), linkedName);
```

It might make the code easier to understand.